internal/impl/devbox.go

```diff
@@ -1190,3 +1176,35 @@ func (d *Devbox) addHashToEnv(env map[string]string) error {
 	}
 	return err
 }
+
+func (d *Devbox) convertEnvToMap(currentEnv []string) (map[string]string, error) {
```

Comment:
This function is doing much more than converting env to map. We should name it accordingly. It sounds like a generic transformation. Always err on the side of over-specifying.